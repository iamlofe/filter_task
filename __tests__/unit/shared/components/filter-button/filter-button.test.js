import React from 'react';
import { shallow, mount } from 'enzyme';

import ButtonFilter from 'shared/components/button-filter/button-filter';

describe('button-filter component', () => {
    const baseProps = {
        onClick: jest.fn(),
        label: 'test',
        className: 'testClass'
    };

    it('should shalow rendered without crashing', () => {
        const filterButton = <ButtonFilter {...baseProps} />;

        const filterButtonComponent = shallow(filterButton);
        expect(filterButtonComponent).toMatchSnapshot();
    });

    it('should have label', () => {
        const filterButton = <ButtonFilter {...baseProps} />;

        const filterButtonComponent = shallow(filterButton);
        expect(filterButtonComponent.text()).toEqual(baseProps.label);
    });
});
