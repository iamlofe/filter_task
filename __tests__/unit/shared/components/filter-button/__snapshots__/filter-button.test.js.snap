// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button-filter component should shalow rendered without crashing 1`] = `
<div
  className="filter-button testClass"
>
  <p
    onClick={[MockFunction]}
  >
    test
  </p>
</div>
`;
